src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -439,4 +504,109 @@ public boolean isSelected() {
             return selected;
         }
     }
+
+    public ProgressiveRendering getGeneratorByTaskId(@QueryParameter String taskId) throws Exception {
+        return generatorByTaskId.get(UUID.fromString(taskId));
+    }
+
+    public static class SupportBundleAsyncGenerator extends ProgressiveRendering {
+        private final Logger logger = Logger.getLogger(SupportBundleAsyncGenerator.class.getName());
+        private UUID taskId;
+        private boolean isCompleted;
+        private String pathToBundle;
+        private List<Component> components;
+        private boolean supportBundleGenerationInProgress = false;
+        private String supportBundleName;
+
+        public SupportBundleAsyncGenerator init(UUID taskId, List<Component> components) {
+            this.taskId = taskId;
+            this.components = components;
+            return this;
+        }
+
+        @Override
+        protected void compute() throws Exception {
+            if (supportBundleGenerationInProgress) {
+                logger.fine("Support bundle generation already in progress, for task id " + taskId);
+                return;
+            }
+            this.supportBundleName = BundleFileName.generate();
+            this.supportBundleGenerationInProgress = true;
+            logger.fine("Generating support bundle... task id " + taskId);
+            File outputDir = new File(SUPPORT_BUNDLE_CREATION_FOLDER + "/" + taskId);
+            if (!outputDir.exists()) {
+                if (!outputDir.mkdirs()) {
+                    throw new IOException("Failed to create directory: " + outputDir.getAbsolutePath());
+                }
+            }
+
+            try (FileOutputStream fileOutputStream = new FileOutputStream(new File(outputDir, supportBundleName))) {
+                SupportPlugin.setRequesterAuthentication(Jenkins.getAuthentication2());
+                try {
+                    try (ACLContext ignored = ACL.as2(ACL.SYSTEM2)) {
+                        SupportPlugin.writeBundle(fileOutputStream, components);
+                    } catch (IOException e) {
+                        logger.log(Level.WARNING, e.getMessage(), e);
```

Comment:
```suggestion
                        logger.log(Level.WARNING, null, e);
```
if you have nothing more informative to say

Reply:
This issues was removed while I addressed https://github.com/jenkinsci/support-core-plugin/pull/617#discussion_r1949699443